StRoot/StETofCalibMaker/StETofCalibMaker.cxx

```diff
@@ -2609,3 +2667,267 @@ StETofCalibMaker::writeHistograms()
+	case 42	:	
+	int tmp       = intVec.at(i) % 4200000000;
+	int stateInt1 = tmp / 10000;
+	int stateInt2 = tmp % 10000;
+	
+	int Get4Id1 = -1;
+	int get4state1 = -1;
+	int Get4Id2 = -1;
+	int get4state2 = -1;
+	
+	if(stateInt1 < 6912){
+	  Get4Id1 = statInt1 % 1728;
+	  get4state1 = stateInt1 / 1728;
+	}
+	if(stateInt2 < 6912){
+	  Get4Id2 = statInt2 % 1728;
+	  get4state2 = stateInt2 / 1728;
+	}
+	
+	if(i < 864){
+	  
+	  mGet4StateMap[Get4Id1] = get4state1;
+	  mGet4StateMap[Get4Id2] = get4state2;
+	  mGet4ZeroStateMap[Get4Id1] = get4state1;
+	  mGet4ZeroStateMap[Get4Id2] = get4state2;
+	}
+	stateVec[lastEvtId].push_back(get4state1);
+	get4IdVec[lastEvtId].push_back(Get4Id1);
+	stateVec[lastEvtId].push_back(get4state2);
+	get4IdVec[lastEvtId].push_back(Get4Id2);
+
+	break;
+		
+        //decode eventnumber ( int = 40.xxx.xxx.xxx = event number ) 
+	case 40:
+		
+	int EvtId = intVec.at(i) % 4000000000;   
+
+	startVec.push_back(EvtId);
+	mMasterStartVec.push_back(EvtId);
+	
+	lastEvtId = EvtId;
+
+	break;		
+		
+        // decode nonZero/stateChange ints ( int = 41.xxx.x00.000 = 1 states only)
+	case 41:
+		
+	int tmp       = intVec.at(i) % 4100000000;
+	int stateInt1 = tmp / 10000;
+	int Get4Id1 = -1;
+	int get4state1 = -1;
+	
+	if(stateInt1 <= 6912) {	 
+	  Get4Id1    = stateInt1 % 1728;
+	  get4state1 = stateInt1 / 1728;
+	}
+	
+	stateVec[lastEvtId].push_back(get4state1);
+	get4IdVec[lastEvtId].push_back(Get4Id1);
+
+	break;
+
+	default:
+	LOG_ERROR << "Get4 state not well defined -> Check db / state file !" << endm;	
+     }
+   }
+
+   // fill stateMap & steering vecs with EvtZero entries: read in first 1728 states & times
+   for(int i = 0; i< 1728;i++){
+     
+     for(unsigned int j=0; j< startVec.size(); j++){
+       
+       unsigned long int key = startVec.at(j);
+       
+       for(unsigned int n =0; n < get4IdVec.at(key).size(); n++){
+	 
+	 //steering vecs
+	 if(i == get4IdVec.at(key).at(n)){ 
+	   mStateVec[i].push_back(stateVec.at(key).at(n));
+	   mStartVec[i].push_back(startVec.at(j));	   
+	 } 	 
+       }  
+     }
+   }
+   
+   //set map validity check evtids ... EvtZero states only valid to first change of state on any get4
+   mStateMapStart = 0 ;
+   mStateMapStop  = startVec.at(0);
+   mDbEntryStart   = startVec.at(0);
+   mDbEntryStop    = startVec.at((startVec.size()-1));
+
+
+   if(fileZero){
+     mDbEntryStart   = 0;
+   }
+  
+   sort( mMasterStartVec.begin(), mMasterStartVec.end() );
+   mMasterStartVec.erase( unique( mMasterStartVec.begin(), mMasterStartVec.end() ), mMasterStartVec.end() );
+
+ }
+
+// -------------------------------------------------------------------------------
+
+void StETofCalibMaker::checkGet4State(unsigned long int eventNr){
+
+  if(eventNr >= mStateMapStart && eventNr < mStateMapStop) {
+    return; // stateMap still valid 
+  }
+
+  unsigned long int closestStop  = 99999999;
+  unsigned long int closestStart = 0;
+
+  //loop over stateMap
+
+  for(unsigned int i =0; i< 1728; i++){
```

Comment:
Avoid using hard coded numbers, please change the limit to something like this: `min(mStartVec.size(), mStateVec.size())`

Reply:
replaced instances of 1728 with eTofConst::nGet4sInSystem